snuba/cli/dlq_manager.py

```diff
@@ -0,0 +1,151 @@
+import os
+from typing import MutableSequence, Optional, Sequence, Tuple
+
+import click
+from arroyo import Message, Partition
+from arroyo import Topic as ArroyoTopic
+from arroyo.backends.kafka import KafkaConsumer, KafkaPayload
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.factory import STORAGES_WITH_DLQ
+from snuba.utils.streams.configuration_builder import build_kafka_consumer_configuration
+
+
+@click.group()
+def dlq_manager() -> None:
+    """
+    Tools for Snuba's Dead Letter Queues
+    """
+    pass
+
+
+@dlq_manager.command()
+@click.option(
+    "--storage",
+    "storage_set",
+    type=click.Choice(
+        [storage_set_key.value for storage_set_key in STORAGES_WITH_DLQ.keys()]
+    ),
+    help="The storage set to list dead letters for",
+    required=True,
+)
+@click.option(
+    "--offset",
+    "offset",
+    type=int,
+    help="An offset to start reading from, 0 by default",
+)
+@click.option(
+    "--limit",
+    "limit",
+    type=int,
+    help="Max number of messages to display, 10 by default",
+)
+def list(storage_set: str, offset: Optional[int], limit: Optional[int]) -> None:
+    """
+    List all messages found in a dead-letter topic
+    """
+    limit = limit or 10
+    offset = offset or 0
+    messages = _consume_dead_letters(storage_set, offset, limit)
+    line_break = "-" * 50
+    if messages:
+        click.echo(
+            f"\nListing the first {limit} messages from "
+            f"offset {offset} in {storage_set} dead-letter topic:"
+            f"\n(see --help for pagination)\n"
+        )
+        for message in messages:
+            click.echo(line_break)
+            click.echo(
+                {
+                    "offset": message.offset,
+                    "invalid_message": message.payload.value,
+                }
+            )
+    else:
+        click.echo(f"\nNo messages found in {storage_set} dead-letter topic!")
+    click.echo(line_break)
+
+
+def _consume_dead_letters(
+    storage_set: str, offset: int, limit: int
+) -> Sequence[Message[KafkaPayload]]:
+    dead_letter_topic_arroyo = ArroyoTopic(
+        STORAGES_WITH_DLQ[StorageSetKey(storage_set)].value
+    )
+    consumer = _build_consumer(storage_set)
+    messages: MutableSequence[Message[KafkaPayload]] = []
+
+    if offset != 0:
+        try:
+            message = consumer.poll(10)
+            consumer.tell()
+            consumer.seek({Partition(dead_letter_topic_arroyo, 0): offset})
```

Comment:
How come hard coding the partition index to 0? What if the dead letter topic has more than one partition?